docs/appendices/release-notes/unreleased.rst

```diff
@@ -243,3 +243,7 @@ Fixes
     UNION
     SELECT i, avg(k), j FROM t3 GROUP BY 1, 3
     LIMIT 10
+
+- Fixed an issue which prevented ``INSERT INTO ... SELECT ...`` from inserting
+  any records if the target table had a partitioned column of a none-string
+  type, used in any expressions: `GENERATED``, ``CHECK``.
```

Comment:
a kept "non-string" since for users it's a passing scenario but can remove it if its' too detailed.